File: backend/src/app/schemas/job_fit.py
from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, Field


class GapItem(BaseModel):
    category: str
    description: str
    severity: str  # high, medium, low
    suggestion: str


class ReorderSuggestion(BaseModel):
    rationale: str
    proposed_order: list[str]


class JobFitResponse(BaseModel):
    id: uuid.UUID
    job_title: str | None
    job_company: str | None
    overall_score: float
    skills_score: float
    experience_score: float
    stack_score: float
    tone_score: float
    gaps: list[GapItem]
    suggestions: list[str]
    reorder_suggestion: ReorderSuggestion | None
    created_at: datetime

    model_config = {"from_attributes": True}


class JobFitHistoryItem(BaseModel):
    id: uuid.UUID
    job_title: str | None
    job_company: str | None
    overall_score: float
    skills_score: float
    experience_score: float
    stack_score: float
    tone_score: float
    created_at: datetime

    model_config = {"from_attributes": True}


class JobFitHistoryResponse(BaseModel):
    analyses: list[JobFitHistoryItem]


class JobFitDeleteResponse(BaseModel):
    message: str


class JobFitRequest(BaseModel):
    job_posting: str = Field(..., max_length=10000)


class DimensionScores(BaseModel):
    skills: float
    experience: float
    stack: float
    tone: float

File: frontend/src/lib/api/job-fit.ts
import { apiClient } from "@/lib/api-client";
import { withAuth } from "@/lib/api/auth-helpers";

export interface GapItem {
  category: string;
  description: string;
  severity: string;
  suggestion: string;
}

export interface ReorderSuggestion {
  rationale: string;
  proposed_order: string[];
}

export interface JobFitResponse {
  id: string;
  job_title: string | null;
  job_company: string | null;
  overall_score: number;
  skills_score: number;
  experience_score: number;
  stack_score: number;
  tone_score: number;
  gaps: GapItem[];
  suggestions: string[];
  reorder_suggestion: ReorderSuggestion | null;
  created_at: string;
}

export interface JobFitHistoryItem {
  id: string;
  job_title: string | null;
  job_company: string | null;
  overall_score: number;
  skills_score: number;
  experience_score: number;
  stack_score: number;
  tone_score: number;
  created_at: string;
}

export interface JobFitHistoryResponse {
  analyses: JobFitHistoryItem[];
}

export const jobFitApi = {
  analyze: (facetId: string, jobPosting: string) =>
    apiClient.post<JobFitResponse>(
      `/api/facets/${facetId}/job-fit`,
      { job_posting: jobPosting },
      withAuth(),
    ),

  getHistory: (facetId: string) =>
    apiClient.get<JobFitHistoryResponse>(
      `/api/facets/${facetId}/job-fit/history`,
      withAuth(),
    ),

  delete: (analysisId: string) =>
    apiClient.delete<{ message: string }>(
      `/api/job-fit/${analysisId}`,
      withAuth(),
    ),
};
